app/Containers/User/Tasks/FindUserByEmailTask.php

```diff
@@ -32,7 +35,7 @@ public function __construct(UserRepository $repository)
     public function run(string $email): User
     {
         try {
-            return $this->repository->findByField('email', $email)->first();
+            return $this->repository->findByField('email', $email)->firstOrFail();
```

Comment:
there is no `firstOrFail` method on the Repository!

Reply:
You can find firstOrFail in vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php line 408.

Reply:
But this is not eloquent

Reply:
What do think would be better, adding the validation rule or changing the findUserByEmailTask to something like this
```
$user = $this->repository->findByField('email', $email)->first();
if (null === $user)
    throw new NotFoundException();
}
```